core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -320,6 +320,80 @@ public static Object getDataBasedOnDataType(String data, DataType actualDataType
 
   }
 
+  public static byte[] getBytesBasedOnDataTypeForNoDictionaryColumn(String dimensionValue,
+      DataType actualDataType) throws Throwable {
+    if (DataType.STRING.equals(actualDataType)) {
```

Comment:
Can you move this to switch case. Is there any reason it can't be moved to switch case?